src/neo-vm/ExecutionEngine.cs

```diff
@@ -1484,6 +1484,8 @@ private void ExecuteStoreToSlot(Slot? slot, int index)
         /// <param name="ex">The exception to be thrown.</param>
         protected void ExecuteThrow(StackItem ex)
         {
+            if (ex is not PrimitiveType)
```

Comment:
Should we be checking to ensure this is a ByteString instead of any primitive type? If the developer throws a boolean or integer, does code up the stack know how to handle that?